### PR TITLE
Detect enclosure as images using file extension

### PR DESCRIPTION
**Pull request by Alkarex**
Support e.g. https://www.sciencesetavenir.fr/atom.xml , which does not include any type hint:

```html
<link rel="enclosure" href="https://example.net/assets/img/ZZZ.jpg" title="YYY" />
```

Useful for https://github.com/FreshRSS/FreshRSS/pull/3805

### CODE REVIEW

File: app/Models/Feed.php [hunk 1]
```diff
@@ -410,7 +410,8 @@ public function loadEntries($simplePie) {
 						$height = $enclosure->get_height();
 						$width = $enclosure->get_width();
 						$length = $enclosure->get_length();
-						if ($medium === 'image' || strpos($mime, 'image') === 0 || ($mime == '' && $length == null && ($width != 0 || $height != 0))) {
+						if ($medium === 'image' || strpos($mime, 'image') === 0 ||
+							($mime == '' && $length == null && ($width != 0 || $height != 0 || preg_match('/[.](gif|jpe?g|png|svg)$/i', $elink)))) {
```
Comment (Frenzie) [inline app/Models/Feed.php]: Should probably also add WebP and that new one based on AV1.

Comment (Alkarex) [reply app/Models/Feed.php]: Done https://github.com/FreshRSS/FreshRSS/pull/3861/commits/5fd913d0c8c56638b2998b1359bac4a0e676d620

**Review by Frenzie — APPROVED** [review]
I forgot about APNG earlier btw, and fwiw there's also BMP.

**Comment by Alkarex** [comment]
> I forgot about APNG earlier btw, and fwiw there's also BMP.

Indeed, and several more, cf. e.g. https://developer.mozilla.org/docs/Web/Media/Formats/Image_types , but this is just a fallback for cases not providing any other type hint

**Pull request merged**